overlord/state/task.go

```diff
@@ -159,3 +163,16 @@ func (t *Task) Get(key string, value interface{}) error {
 	t.state.ensureLocked()
 	return t.data.get(key, value)
 }
+
+// WaitFor registers another task as a requirement for t to make progress
+// and sets the status as WaitingStatus.
+func (t *Task) WaitFor(another *Task) {
+	t.state.ensureLocked()
+	t.status = WaitingStatus
+	t.waitTasks = append(t.waitTasks, another.ID())
+}
```

Comment:
@mvo5 that was a bit unhelpful, what I did in change was using a map[sting]bool instead of a list to keep a task id sets, as Gustavo remarked we probably still want to serialize it as a list though I could fix both together though (there's a micro task about that in trello)